### PR TITLE
docs(jsx-runtime): improve usage guide and fix invalid links to sb docs

### PR DESCRIPTION
**Pull request by Hotell**
## New Behavior

- see PR title
- addresses https://github.com/microsoft/fluentui/pull/32439/files#r1751562920, https://github.com/microsoft/fluentui-contrib/pull/217#discussion_r1753686793

## Related Issue(s)

- Fixes #

### CODE REVIEW

File: packages/react-components/react-jsx-runtime/README.md [hunk 1]
```diff
@@ -35,15 +40,27 @@ const renderButton_unstable = (state: ButtonState) => {
...
+> 💡 NOTE: no need to specify `@jsx` pragma because our api exports `createElement` - same name symbol as React exports
+
+```tsx
+/** @jsxRuntime classic */
 
-In TSC case you can simply add this do `tsconfig.json` file:
+import { createElement } from '@fluentui/react-jsx-runtime';
+import { assertSlots } from '@fluentui/react-utilities';
 
-```json
```
Comment (Hotell) [inline packages/react-components/react-jsx-runtime/README.md:42]: removed this completely as we should not encourage to use this pragma within every jsx in user land

**Comment by fabricteam** [comment]
## 📊 Bundle size report

✅ No changes found
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)

  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32838/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">36</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32838/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">36</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32838/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.regression.txt">Possible regression</a></td>
           </tr></table><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32838/perf-test-react-components/Avatar-mount_base.html">660</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32838/perf-test-react-components/Avatar-mount.html">624</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32838/perf-test-react-components/Button-mount_base.html">314</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32838/perf-test-react-components/Button-mount.html">311</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32838/perf-test-react-components/Field-mount_base.html">1152</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32838/perf-test-react-components/Field-mount.html">1152</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32838/perf-test-react-components/FluentProvider-mount_base.html">712</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32838/perf-test-react-components/FluentProvider-mount.html">715</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32838/perf-test-react-components/FluentProviderWithTheme-mount_base.html">83</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32838/perf-test-react-components/FluentProviderWithTheme-mount.html">98</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32838/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">36</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32838/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">36</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32838/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32838/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">74</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32838/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">74</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32838/perf-test-react-components/MakeStyles-mount_base.html">851</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32838/perf-test-react-components/MakeStyles-mount.html">864</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32838/perf-test-react-components/Persona-mount_base.html">1758</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32838/perf-test-react-components/Persona-mount.html">1709</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32838/perf-test-react-components/SpinButton-mount_base.html">1389</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32838/perf-test-react-components/SpinButton-mount.html">1334</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32838/perf-test-react-components/SwatchPicker-mount_base.html">1695</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32838/perf-test-react-components/SwatchPicker-mount.html">1664</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: change/@fluentui-react-jsx-runtime-5bfd81c5-626e-4d33-a616-fe50df609366.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-jsx-runtime-5bfd81c5-626e-4d33-a616-fe50df609366.json:1]: ## 🕵🏾‍♀️ [<b data-uid='vrt-gist'></b> visual regressions to review in the fluentuiv9 Visual Regression Report](https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=32838&commitId=05a10321c4520c68d024eff763b95a50cc3f9b0e&env=prod&clientType=FLUENTUI)


  <details><summary><b>Avatar Converged</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Avatar Converged.Badge Mask RTL.chromium.png|7| Changed|

</details>


  <details><summary><b>Drawer</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Drawer.Full Overlay Dark Mode.chromium.png|996| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

File: packages/react-components/react-jsx-runtime/README.md [hunk 3]
```diff
@@ -2,22 +2,28 @@
...
 ## Usage
 
-This library should only be used in the case where you are trying to use the internal slot API of Fluent UI React. If you are not using the internal slot API, you should not need to use this library.
+> [!NOTE]
+> This custom JSX pragma should only be used in cases where you are trying to use the internal Fluent UI React **slot API in conjunction with `assertSlots()`**.
+>
+> If you are not using the internal slot API, don't use it.
 
-In case you want to re-compose a component and redeclare its render method then this API will be necessary, here's our documentation on [Rendering a component with slots](https://react.fluentui.dev/?path=/docs/concepts-developer-customizing-components-with-slots--page#rendering-components-with-slots)
+In case you want to re-compose a component and redeclare its render method then this API will be necessary, learn more on [Rendering a component with slots](https://react.fluentui.dev/?path=/docs/concepts-developer-customizing-components-with-slots--docs#rendering-components-with-slots)
```
Comment (Hotell) [inline packages/react-components/react-jsx-runtime/README.md:14]: @bsunderhus  those linked SB docs doesn't describe anything related to "re-composing/declaring" besides  standard render "hook" implementation pattern. 

I'd rather update the wording here ( remove this completely ) or update the SB docs with examples of actual recomposition pattern. I prefer the latter.

thoughts?

**Comment by bsunderhus** [comment]
> as I was validating assumptions that led to this PR I ran into some use-cases which I'd like to get answers to (@bsunderhus 🙏):
> 
> ## unnecessary use of jsx pragma
> is following as expected or a programatic mistake ? My assumption is that if one doesn't use imperatively `JSXMemberExpression` aka ( `<state.*>` ) no custom pragma or `assertSlots` called are necessary, correct ?
> 
> > src: https://github.com/microsoft/fluentui/blob/master/packages/react-components/react-teaching-popover/library/src/components/TeachingPopoverSurface/renderTeachingPopoverSurface.tsx
> 
> image

You're completely right @Hotell, if there's not slot JSX declaration `<state.*>` there's no need to declare the pragma comment at all. This looks like a ctrl-c ctrl-v case perhaps (as most of the `render*` files usually have it.

File: packages/react-components/react-jsx-runtime/README.md (hunk 3 above)
Comment (bsunderhus) [reply packages/react-components/react-jsx-runtime/README.md:14]: This entire documentation needs some 💖 with more examples of actual use cases. I agree that the latter would be ideal.

**Review by bsunderhus — APPROVED** [review]

File: packages/react-components/react-jsx-runtime/README.md (hunk 3 above)
Comment (Hotell) [reply packages/react-components/react-jsx-runtime/README.md:14]: kk I guess we have a follow up task then ! :D ty

**Pull request merged**